File: server/forcutbook/src/main/java/konkuk/forcutbook/diary/DiaryRepository.java
package konkuk.forcutbook.diary;

import konkuk.forcutbook.diary.domain.Diary;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface DiaryRepository extends JpaRepository<Diary, Long> {
    List<Diary> findByWriterId(Long userId);

    @EntityGraph(attributePaths = "diaryImages")
    Optional<Diary> findById(Long id);
}

File: android/app/src/main/java/com/fourcutbook/forcutbook/data/repository/DiaryRepository.kt
package com.fourcutbook.forcutbook.data.repository

import com.fourcutbook.forcutbook.domain.Diary
import java.io.File

interface DiaryRepository {

    suspend fun fetchDiaries(userId: Long? = null): List<Diary>

    suspend fun fetchDiaryDetails(diaryId: Long): Diary

    suspend fun postImage(image: File): Diary

    // todo: diary와 image file이 각각의 인자로 필요할까?
    suspend fun postDiary(diary: Diary, image: File)
}
